src/popup/router/pages/Retip.vue

```diff
@@ -34,69 +37,97 @@
+    ...mapState('fungibleTokens', ['selectedToken']),
+    ...mapState(['balance', 'tippingV1', 'tippingV2', 'sdk']),
     urlStatus() {
       return this.$store.getters['tipUrl/status'](this.tip.url);
     },
+    shouldUseV2Contract() {
+      return this.$route.query.id.includes('_v2') || this.$route.query.id.includes('_v3');
+    },
   },
   watch: {
     amount() {
-      this.amountError = !+this.amount || this.amount < this.minTipAmount;
+      this.amountError = !+this.amount || (!this.selectedToken && this.amount < this.minTipAmount);
     },
   },
   async created() {
     this.loading = true;
     await this.$watchUntilTruly(() => this.sdk);
     this.minCallFee = calculateFee(TX_TYPES.contractCall, {
       ...this.sdk.Ae.defaults,
-      contractId: this.activeNetwork.tipContractV1,
+      contractId: this.shouldUseV2Contract
+        ? this.activeNetwork.tipContractV2
+        : this.activeNetwork.tipContractV1,
       callerId: this.account.publicKey,
     }).min;
-    await this.$watchUntilTruly(() => this.tippingV1);
-    const tipId = +this.$route.query.id;
+    await this.$watchUntilTruly(() => this.tippingV1, this.tippingV2);
+    const tipId = this.$route.query.id;
     if (!tipId) throw new Error('"id" param is missed');
-    const { decodedResult } = await this.tippingV1.methods.get_state();
-    this.tip = tipping.getTipsRetips(decodedResult).tips.find(({ id }) => id === tipId);
+    this.tippingContract = this.shouldUseV2Contract ? this.tippingV2 : this.tippingV1;
+    const state = await this.tippingContract.methods.get_state();
+    const tipsAndRetips = await tipping.getTipsRetips(state);
+    this.tip = tipsAndRetips.tips.find(({ id }) => id === tipId);
     this.loading = false;
   },
   methods: {
     async sendTip() {
       const calculatedMaxValue =
         this.balance > this.minCallFee ? this.balance - this.minCallFee : 0;
       this.amountError = !this.amount || !this.minCallFee || calculatedMaxValue - this.amount <= 0;
-      this.amountError = this.amountError || !+this.amount || this.amount < this.minTipAmount;
+      this.amountError =
+        this.amountError ||
+        !+this.amount ||
+        (!this.selectedToken && this.amount < this.minTipAmount);
       if (this.amountError) return;
-      const amount = BigNumber(this.amount).shiftedBy(MAGNITUDE);
+      const amount = this.selectedToken
+        ? convertToken(this.amount, this.selectedToken.decimals).toFixed()
+        : convertToken(this.amount, MAGNITUDE).toFixed();
       this.loading = true;
       try {
-        const { hash } = await this.tippingV1.methods.retip(this.tip.id, {
-          amount,
-          waitMined: false,
-          modal: false,
-        });
-        if (hash) {
+        let retipResponse = null;
+        if (this.selectedToken) {
+          await this.$store.dispatch('fungibleTokens/createOrChangeAllowance', this.amount);
+          retipResponse = await this.tippingV2.methods.retip_token(
```

Comment:
Don't you want to add `waitMined: false, modal: false` to be consistent with the `retip` call from below? Actually, I don't know what `modal: false` means.

Reply:
The retip_token call should follow exact parameters structure as it is.

Reply:
These parameters are handled by sdk/wallet, you should consider them to have a consistent behavior with a usual retip call.